### DISCUSSION TITLE
How to consider coord_scalar

### DISCUSSION THREAD

**Discussion by khaled1240274** [trhallam/segysak#136] [Q&A] [answered]
I need your help for implementing `scale_coords(coord_scalar=None)` and `coord_scalar_mult = np.power(abs(coord_scalar), np.sign(coord_scalar))`
My segy cube textual header says that "**X-Coord, 4I, 181-184   Y-Coord, 4I, 185-188 (Both Multipled by 100)**"
I want to know how to consider this while reading the segy file
`segy = xr.open_dataset(
                file_path, engine='sgy_engine',
                dim_byte_fields={"iline": InlineLoc, "xline": XlineLoc},
                extra_byte_fields={"cdp_x": cdpxLoc, "cdp_y": cdpyLoc}
            )`
Specially it gives wrong data when try to extract trace along the well path

**Comment by trhallam** [comment, marked as answer]
Hi, coordinate scaling is not handled by default and must be applied by the user.

Any trace header information must be stored as intergers which is why, if you have floating point precission coordinates you must apply a scaling factor. To scale your coordinates, apply the following

```python
segy = xr.open_dataset(
    file_path, engine='sgy_engine', 
    dim_byte_fields={"iline": InlineLoc, "xline": XlineLoc}, 
    extra_byte_fields={"cdp_x": cdpxLoc, "cdp_y": cdpyLoc} 
)
# multiply by 100
segy.segysak.scale_coords(100)
# multiply by 0.01
# segy.segysak.scale_coords(-100)
```

The convention here is part of the SEG-Y standard and comes from how the coordinate scalar (also an integer) is represented in the trace headers.

You can find the reference for this function [here](https://trhallam.github.io/segysak/latest/api/xarray_new.html#segysak._accessor.SegysakDatasetAccessor.scale_coords) but I know the Table for the function definition from the [docstring](https://github.com/trhallam/segysak/blob/7a85c1b4da3ea4d10c612dd36220377267c3f8e5/segysak/_accessor.py#L455) doesn't render well in the HTML reference..

**Reply by khaled1240274** [reply]
Thanks a lot.
